mailpoet/tests/DataFactories/AutomationRun.php

```diff
@@ -0,0 +1,155 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Test\DataFactories;
+
+use DateTimeImmutable;
+use MailPoet\Automation\Engine\Data\Automation;
+use MailPoet\Automation\Engine\Data\AutomationRun as Entity;
+use MailPoet\Automation\Engine\Data\AutomationRunLog;
+use MailPoet\Automation\Engine\Data\NextStep;
+use MailPoet\Automation\Engine\Data\Step;
+use MailPoet\Automation\Engine\Data\Subject;
+use MailPoet\Automation\Engine\Storage\AutomationRunLogStorage;
+use MailPoet\Automation\Engine\Storage\AutomationRunStorage;
+use MailPoet\Automation\Engine\Storage\AutomationStorage;
+use MailPoet\DI\ContainerWrapper;
+
+class AutomationRun {
+
+
+  /** @var ?Entity */
+  private $automationRun = null;
+
+  /** @var Automation  */
+  private $automation;
+
+  /** @var array Subject[] */
+  private $subjects = [];
+
+  /** @var string  */
+  private $triggerKey = '';
+
+  /** @var \DateTimeImmutable | null */
+  private $createdAt;
+
+  /** @var \DateTimeImmutable | null */
+  private $updatedAt;
+
+  /** @var string  */
+  private $status = Entity::STATUS_COMPLETE;
+
+  /** @var ?string */
+  private $nextStep = null;
+
+  private $storage;
+
+  public function __construct() {
+    $this->storage = ContainerWrapper::getInstance(WP_DEBUG)->get(AutomationRunStorage::class);
+  }
+
+  public function withAutomation(Automation $automation): self {
+    $this->automation = $automation;
+    return $this;
+  }
+
+  public function withSubject(Subject $subject): self {
+    $this->subjects[] = $subject;
+    return $this;
+  }
+
+  public function withTriggerKey(string $triggeKey): self {
+    $this->triggerKey = $triggeKey;
+    return $this;
+  }
+
+  public function withCreatedAt(DateTimeImmutable $createdAt): self {
+    $this->createdAt = $createdAt;
+    if (!$this->updatedAt) {
+      return $this->withUpdatedAt($createdAt);
+    }
+    return $this;
+  }
+
+  public function withUpdatedAt(DateTimeImmutable $updatedAt): self {
+    $this->updatedAt = $updatedAt;
+    if (!$this->createdAt) {
+      return $this->withCreatedAt($updatedAt);
+    }
+    return $this;
+  }
+
+  public function withStatus(string $status): self {
+    $this->status = $status;
+    return $this;
+  }
+
+  public function withNextStep(string $nextStep = null): self {
+    $this->nextStep = $nextStep;
+    return $this;
+  }
+
+  public function create(): Entity {
+
+    $automationRun = new Entity(
+      $this->automation->getId(),
+      $this->automation->getVersionId(),
+      $this->triggerKey,
+      $this->subjects
+    );
+
+    $id = $this->storage->createAutomationRun($automationRun);
+    $automationRun = $this->storage->getAutomationRun($id);
+    $this->storage->deleteAutomationRun($automationRun);
```

Comment:
Why does this create and immediately delete an automation run? 🤔

Reply:
Fixed in https://github.com/mailpoet/mailpoet/pull/5065/commits/f7f46c57c6bf0df1db31dfa125c1481b605e701a.